system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/AbstractEndToEndTransfer.java

```diff
@@ -48,8 +48,7 @@ void httpPullDataTransfer() {
         var assetId = UUID.randomUUID().toString();
         createResourcesOnProvider(assetId, noConstraintPolicy(), UUID.randomUUID().toString(), httpDataAddressProperties());
 
-        var catalog = CONSUMER.getCatalog(PROVIDER.idsEndpoint());
-        assertThat(catalog.getContractOffers()).hasSizeGreaterThan(0);
```

Comment:
Not a big fan of moving test assertions into util methods, because the make the test code less readable and tedious to maintain. 
If there are recurring assertions, at the very least I would create a dedicated static method in some test util class with a descriptive name.

Reply:
Generally I agree with you, but these are end to end tests, they should demonstrate the system from an hi-level perspective, showing everything in the main test method will make the test pretty impossible to read.
These assertions are not meaningful to the scope of the test, but they verifies that no regression are going to be introduced on some really important details.

Reply:
I'd argue that guarding against regression should be a separate, explicit test then (IDK if we have one already), where the assertion is written out.
if it is outside of the scope, the assertion is not necessary...

let's not get hung up on that though.